tests/tokenWrapper.test.ts

```diff
@@ -0,0 +1,800 @@
+
+        const {tokenName, tokenSymbol, decimal, contractGovernor, feeRecipient, feePercentage, isNativeAllowed, wrappingLimit,
+            contractProposalNonce, tokenAddress, totalSupply, governorBalance} = tokenWrapperContractInitParams(sender, BobSigner, CharlieSigner)
+
+
+        // token wrapper instantiation
+        const tokenWrapperContractFactory = await getContractFactory('governed_token_wrapper', sender.address);
+
+        const tokenWrapperContract = await tokenWrapperContractFactory.deploy('new',
+            tokenName, tokenSymbol, decimal, contractGovernor, feeRecipient, feePercentage, isNativeAllowed, wrappingLimit, contractProposalNonce, tokenAddress,
+            totalSupply, governorBalance);
+
+        return { sender, Alice, Charlie, CharlieSigner, Bob, BobSigner, Ferdie, FerdieSigner, Eve, EveSigner, Dave, DaveSigner,
+            tokenWrapperContractFactory, tokenWrapperContract};
+    }
+
+    function tokenWrapperContractInitParams(sender: any, BobSigner: any, CharlieSigner: any) {
+        let tokenName = 'Webb';
+        let tokenSymbol = 'Webb';
+        let decimal = 4;
+        let contractGovernor = sender.address;
+        let feeRecipient = sender.address;
+        let feePercentage = 1;
+        let isNativeAllowed = true;
+        let wrappingLimit = 10;
+        let contractProposalNonce = 0;
+        let tokenAddress = BobSigner.address
+        let totalSupply = 1_000_000_000_000_000;
+        let governorBalance = 9_000_000;
+
+
+        return {tokenName, tokenSymbol, decimal, contractGovernor, feeRecipient, feePercentage, isNativeAllowed, wrappingLimit,
+            contractProposalNonce, tokenAddress, totalSupply, governorBalance}
+    }
+
+    it('Add token address', async () => {
+        const {tokenName, contractProposalNonce} = tokenWrapperContractInitParams(sender, BobSigner, CharlieSigner)
+
+        let governor = await tokenWrapperContract.query.governor();
+        let name = await tokenWrapperContract.query.name();
+
+        console.log(`governor is ${governor.output}`);
+        console.log(`token name is ${name.output}`);
+
+        expect(governor.output).to.equal(sender.address);
+        expect(name.output).to.equal(tokenName);
+
+        let signer = tokenWrapperContract.signer;
+
+        console.log(`signer is ${signer}`);
+
+        expect( await tokenWrapperContract.tx.addTokenAddress(BobSigner.address, contractProposalNonce + 1)).to.be.ok;
+
+        // validate that address has been added successfully
+        let  isValidAddress = await tokenWrapperContract.query.isValidTokenAddress(BobSigner.address);
+        expect(isValidAddress.output).to.equal(true);
+
+        // validate that proposalNonce has increased
+        let  newProposalNonce = await tokenWrapperContract.query.nonce();
+        expect(newProposalNonce.output).to.be.equal(contractProposalNonce + 1);
+    });
+
+    it('Remove token address', async () => {
```

Comment:
The test here as well uses `BobSigner.address` as a token to add/remove from the list. This is not exactly the most clear to an outside observer nor does it demonstrate the intended usage of such contract functions. I would suggest using a real token address so that the test is clear.